### PR TITLE
Rem 348

### PR DESCRIPTION
**Pull request by FatimaSalami**
this task is to reset the pass from login page.
first you got the enter userName form where a code will be sent to your email you write here.
Then another form which requires the code and the new pass with its verification will open saving thus the new password and taking you back to login page.
You can check the behaviour currently in Argus asp accounts.
The file containing the forms is separate (i am not opening now the reset enter userName form from login.. i am opening new file in new path by clicking the forget pass link in login page..)

### CODE REVIEW

File: src/pages/forget-password/forms/ResetChPassForm.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          .post(`${process.env.NEXT_PUBLIC_AuthURL}MA.asmx/resetPW`, bodyFormData)
+          .then(res => {
+            toast.success('Password changed successfully!')
+            router.push('/login')
+          })
+          .catch(error => {
+            stackError({ message: error })
+          })
+      } else {
+        toast.error('Passwords do not match!')
```
Comment (RebeccaFrem02) [inline src/pages/forget-password/forms/ResetChPassForm.js:57]: can call this from platformLabels?

File: src/pages/forget-password/forms/ResetChPassForm.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          code: formik.values.code
+        }
+
+        var bodyFormData = new FormData()
+        bodyFormData.append('record', JSON.stringify(resetPWVal))
+
+        axios
+          .post(`${process.env.NEXT_PUBLIC_AuthURL}MA.asmx/resetPW`, bodyFormData)
+          .then(res => {
+            toast.success('Password changed successfully!')
```
Comment (RebeccaFrem02) [inline src/pages/forget-password/forms/ResetChPassForm.js:50]: can call this from platformLabels?

File: src/pages/forget-password/reset.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                <Grid item xs={12} container justifyContent='flex-end'>
+                  <Button size='small' type='submit' variant='contained' sx={{ mb: 7 }} onClick={formik.handleSubmit}>
+                    {platformLabels.Reset}
+                  </Button>
+                </Grid>
+              </Grid>
+            </CardContent>
+          </Card>
+
+          {/* Footer Section */}
```
Comment (RebeccaFrem02) [inline src/pages/forget-password/reset.js:145]: is this comment necessary

File: src/pages/forget-password/forms/ResetChPassForm.js (hunk 1 above)
Comment (FatimaSalami) [reply src/pages/forget-password/forms/ResetChPassForm.js]: changed in this and in changePass form you already created also

File: src/pages/forget-password/forms/ResetChPassForm.js (hunk 2 above)
Comment (FatimaSalami) [reply src/pages/forget-password/forms/ResetChPassForm.js]: changed in this and in changePass form you already created also

File: src/pages/forget-password/reset.js (hunk 3 above)
Comment (FatimaSalami) [reply src/pages/forget-password/reset.js]: removed

File: src/pages/forget-password/forms/ResetPassForm.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      } else {
+        toast.error(labels.passNotMatching)
+      }
+    }
+  })
+
+  return (
+    <VertLayout>
+      <Grow>
+        <Grid container spacing={3} sx={{ pl: '10px', pt: '10px', pr: '10px' }}>
```
Comment (Ghinwa-Arab) [inline src/pages/forget-password/forms/ResetPassForm.js:65]: instead pl: '10px', pr: '10px' use px:'10px'

px is shorthand for setting both pl and pr

Comment (Ghinwa-Arab) [inline src/pages/forget-password/forms/ResetPassForm.js:65]: is using 2 grids here neccessary? it's possible to set directly customTextField inside grow?

File: src/pages/forget-password/forms/ResetPassForm.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import * as yup from 'yup'
+import { AuthContext } from 'src/providers/AuthContext'
+import { useAuth } from 'src/hooks/useAuth'
+import { useError } from 'src/error'
+import NewPassword from 'src/components/Shared/NewPassword'
+import axios from 'axios'
+import { useRouter } from 'next/router'
+
+const ResetPassForm = ({ labels, username = '' }) => {
+  const [score, setScore] = useState(0)
```
Comment (Ghinwa-Arab) [inline src/pages/forget-password/forms/ResetPassForm.js:17]: Why are we declaring the state here if it's only used in the NewPassword ? Wouldn't it be better to declare it inside the NewPassword component itself?

File: src/pages/forget-password/reset.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+const Reset = () => {
+  const theme = useTheme()
+  const { platformLabels } = useContext(ControlContext)
+  const [accountId, setAccountId] = useState({})
+  const { stack } = useWindow()
+  const auth = useAuth()
+
+  useEffect(() => {
+    setAccountId(auth?.getAC?.data?.record?.accountId)
```
Comment (Ghinwa-Arab) [inline src/pages/forget-password/reset.js:25]: can we remove account state, useEffect and replace instead
const accountId = auth?.getAC?.data?.record?.accountId  ???

File: src/pages/forget-password/forms/ResetPassForm.js (hunk 4 above)
Comment (FatimaSalami) [reply src/pages/forget-password/forms/ResetPassForm.js:65]: i need the padding in grid 
if they are removed display will be affected

File: src/pages/forget-password/forms/ResetPassForm.js (hunk 5 above)
Comment (FatimaSalami) [reply src/pages/forget-password/forms/ResetPassForm.js:17]: this is a concept done already by rebecca in newpassword 
and i followed it.. 
i guess in the future modifications will be done 
but this is the initial version for now

File: src/pages/forget-password/reset.js (hunk 6 above)
Comment (FatimaSalami) [reply src/pages/forget-password/reset.js:25]: right fixed

File: src/pages/forget-password/forms/ResetPassForm.js (hunk 5 above)
Comment (hadichahine) [inline src/pages/forget-password/forms/ResetPassForm.js:17]: Move this inside NewPassword. Fix this inside other pages using NewPassword too

File: src/pages/login/index.js [hunk 7]
```diff
@@ -209,7 +209,7 @@ const LoginPage = () => {
                   justifyContent: 'space-between'
                 }}
               >
-                <LinkStyled href='/pages/auth/forgot-password-v1'>{platformLabels.ForgotPass}</LinkStyled>
+                <LinkStyled href='../../forget-password/reset'>{platformLabels.ForgotPass}</LinkStyled>
```
Comment (hadichahine) [inline src/pages/login/index.js:212]: why the ../../

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Pull request merged**